nix/docker/context/bin/run-node

```diff
@@ -0,0 +1,276 @@
+CNODE_HOSTNAME="$CARDANO_PUBLIC_IP"
+CNODE_PORT=$CARDANO_PORT
+CUSTOM_PEERS="$CARDANO_CUSTOM_PEERS"
+GENESIS_JSON="$CARDANO_CONFIG_BASE/mainnet-shelley-genesis.json"
+TOPOLOGY="$CARDANO_TOPOLOGY"
+LOG_DIR="$CARDANO_LOG_DIR"
+EKG_HOST="$EKG_HOST"
+EKG_PORT=$EKG_PORT
+EKG_TIMEOUT=$EKG_TIMEOUT
+MAX_PEERS=$MAX_PEERS
+
+# Mapping for gLiveView
+CCLI="cardano-cli"
+CNODE_PORT=$CARDANO_PORT
+CONFIG="$CARDANO_CONFIG"
+TOPOLOGY="$CARDANO_TOPOLOGY"
+SOCKET="$CARDANO_SOCKET_PATH"
+LOG_DIR="$CARDANO_LOG_DIR"
+EOF
+}
+
+#####################################################################
+#
+# Run the relay node in the background
+#
+runRelayNode () {
+
+  effopts=(--config $CARDANO_CONFIG \
+    --topology $CARDANO_TOPOLOGY \
+    --database-path $CARDANO_DATABASE_PATH \
+    --socket-path $CARDANO_SOCKET_PATH \
+    --host-addr $CARDANO_BIND_ADDR \
+    --port $CARDANO_PORT)
+
+  effopts+=(${options[@]})
+
+  echo "cardano-node run ${effopts[@]}"
+  cardano-node run ${effopts[@]} &
+}
+
+#####################################################################
+#
+# Run the block producer in the background
+#
+runBlockProducerNode () {
+
+  effopts=(--config $CARDANO_CONFIG \
+    --topology $CARDANO_TOPOLOGY \
+    --database-path $CARDANO_DATABASE_PATH \
+    --socket-path $CARDANO_SOCKET_PATH \
+    --host-addr $CARDANO_BIND_ADDR \
+    --port $CARDANO_PORT \
+    --shelley-kes-key $CARDANO_SHELLEY_KES_KEY \
+    --shelley-vrf-key $CARDANO_SHELLEY_VRF_KEY \
+    --shelley-operational-certificate $CARDANO_SHELLEY_OPERATIONAL_CERTIFICATE)
+
+  effopts+=(${options[@]})
+
+  echo "cardano-node run ${effopts[@]}"
+  cardano-node run ${effopts[@]} &
+}
+
+# Override default values with explicit options
+options=($@)
+for i in "${!options[@]}"
+do
+    j=$((i + 1))
+    key=${options[i]}
+    val=${options[j]}
+    found=false
+
+    # echo "$i/$j: ${key} ${val}"
+
+    case ${key} in
+      --config) CARDANO_CONFIG=${val}; found=true;;
+      --topology) CARDANO_TOPOLOGY=${val}; found=true;;
+      --database-path) CARDANO_DATABASE_PATH=${val}; found=true;;
+      --socket-path) CARDANO_SOCKET_PATH=${val}; found=true;;
+      --host-addr) CARDANO_BIND_ADDR=${val}; found=true;;
+      --port) CARDANO_PORT=${val}; found=true;;
+      --shelley-kes-key) CARDANO_SHELLEY_KES_KEY=${val}; found=true;;
+      --shelley-vrf-key) CARDANO_SHELLEY_VRF_KEY=${val}; found=true;;
+      --shelley-operational-certificate) CARDANO_SHELLEY_OPERATIONAL_CERTIFICATE=${val}; found=true;;
+    esac
+
+    if [[ $found == true ]]; then
+      options[i]="";
+      options[j]="";
+    fi
+done
+
+printRunEnv
+writeRootEnv
+
+# The IPC socket dir is not created on demand
+mkdir -p `dirname $CARDANO_SOCKET_PATH`
+
+# Removing a potentialy existing lost+found directory
+rm -rf $CARDANO_DATABASE_PATH/lost+found
+
+if [[ $CARDANO_BLOCK_PRODUCER == true ]]; then
+
+  runBlockProducerNode
+
+else
+
+  if [[ $CARDANO_UPDATE_TOPOLOGY == true ]]; then
+    topologyUpdate -l &
```

Comment:
why is this run in the background? The relay node could start before topology is fully updated.